pkg/ccl/streamingccl/streamproducer/stream_lifetime.go

```diff
@@ -69,7 +74,36 @@ func startReplicationProducerJob(
 	if err != nil {
 		return streampb.ReplicationProducerSpec{}, err
 	}
-	tenantID := tenantRecord.ID
+	tenantID, err := roachpb.MakeTenantID(tenantRecord.ID)
+	if err != nil {
+		return streampb.ReplicationProducerSpec{}, err
+	}
+
+	var replicationStartTime hlc.Timestamp
+	if !req.ReplicationStartTime.IsEmpty() {
+		if tenantRecord.PreviousSourceTenant != nil {
+			cid := tenantRecord.PreviousSourceTenant.ClusterID
+			if !req.ClusterID.Equal(uuid.UUID{}) && !cid.Equal(uuid.UUID{}) {
+				if !req.ClusterID.Equal(cid) {
+					return streampb.ReplicationProducerSpec{}, errors.Errorf("requesting cluster ID %s does not match previous source cluster ID %s",
+						req.ClusterID, cid)
+				}
+			}
+
+			tid := tenantRecord.PreviousSourceTenant.TenantID
+			if !req.TenantID.Equal(roachpb.TenantID{}) && !tid.Equal(roachpb.TenantID{}) {
+				if !req.TenantID.Equal(tid) {
+					return streampb.ReplicationProducerSpec{}, errors.Errorf("requesting tenant ID %s does not match previous source tenant ID %s",
+						req.TenantID, tid)
+				}
+			}
+		}
+		replicationStartTime = req.ReplicationStartTime
```

Comment:
if tenantRecord.PreviousSourceTenant == nil when req.ReplicationStartTime is not empty, we should `errors.AssertionFailed` ?